packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/tests/reducer.test.js

```diff
@@ -1600,6 +1600,95 @@ describe('CONTENT MANAGER | COMPONENTS | EditViewDataManagerProvider | reducer',
         expect(reducer(state, action)).toEqual(expected);
       });
     });
+
+    it('should merge modifiedData with relation containing fields if the modifiedData exists', () => {
+      const state = {
+        ...initialState,
+        formErrors: true,
+        initialData: {},
+        modifiedData: {
+          relation: [
+            {
+              id: 1,
+            },
+          ],
+          componentWithRelation: {
+            relation: [
+              {
+                id: 1,
+              },
+              {
+                id: 2,
+              },
+            ],
+          },
+        },
+        modifiedDZName: true,
+        shouldCheckErrors: true,
+      };
+      const expected = {
+        ...initialState,
+        formErrors: {},
+        initialData: {
+          ok: true,
+          relation: [
+            {
+              id: 1,
+            },
+          ],
+          componentWithRelation: {
+            relation: [
+              {
+                id: 1,
+              },
+              {
+                id: 2,
+              },
+            ],
+            id: 1,
+          },
+        },
+        modifiedData: {
+          ok: true,
+          relation: [
+            {
+              id: 1,
+            },
+          ],
+          componentWithRelation: {
+            relation: [
+              {
+                id: 1,
+              },
+              {
+                id: 2,
+              },
+            ],
+            id: 1,
+          },
+        },
+        modifiedDZName: null,
+        shouldCheckErrors: false,
+      };
+
+      const action = {
+        type: 'INIT_FORM',
+        initialValues: {
+          ok: true,
+          relation: { count: 10 },
+          componentWithRelation: {
+            id: 1,
+            relation: {
+              count: 10,
+            },
+          },
+        },
+        relationalFieldPaths: ['relation', 'componentWithRelation.relation'],
+        componentPaths: ['componentWithRelation'],
+      };
+
+      expect(JSON.stringify(reducer(state, action))).toEqual(JSON.stringify(expected));
```

Comment:
Why do you need to stringify the expected value?

Reply:
I ran into this – https://github.com/facebook/jest/issues/8475

Presumably because i have arrays with data in them 🤔 happy to change it if you know a better way?

Reply:
Mh, I see. Maybe it would be nicer to use `expect.objectContaining()` if the two attributes are part of the response instead of checking the full response?

Reply:
ive tweaked it